public/app/plugins/panel/canvas/editor/element/utils.ts

```diff
@@ -85,8 +93,32 @@ const getData = (api: APIEditorConfig) => {
   return data;
 };
 
-const requestMatchesGrafanaOrigin = (requestEndpoint: string) => {
+const getEndpoint = (endpoint: string) => {
+  const isRelativeUrl = endpoint.startsWith('/');
+  if (isRelativeUrl) {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const sanitizedRelativeURL = textUtil.sanitizeUrl(endpoint) as RelativeUrl;
+    endpoint = createAbsoluteUrl(sanitizedRelativeURL, []);
+  }
+
+  return endpoint;
+};
+
+const isRequestUrlAllowed = (method: HttpRequestMethod, requestEndpoint: string) => {
+  const allowedMethods = [HttpRequestMethod.POST, HttpRequestMethod.PUT];
+  const allowedAPIRegex = new RegExp(allowedAPIEndpointPattern);
+
+  if (allowedMethods.includes(method) && config.actions.allowPostURL !== '') {
+    const allowedRegexFromConfig = new RegExp(config.actions.allowPostURL!);
+    return allowedRegexFromConfig.test(requestEndpoint) && allowedAPIRegex.test(requestEndpoint);
+  }
+
+  if (method === HttpRequestMethod.GET) {
+    return true;
+  }
```

Comment:
This logic doesn't seem to match the description of the intent or maintain backward compatibility with the existing logic.

Since the meaning of the return value for the new `isRequestUrlAllowed` function in inverted compared to the old `requestMatchesGrafanaOrigin`, we should invert the check on line 123 to use `!==`. Right now when `allowedPostURL` is empty we're allowing requests to Grafana endpoints and disallowing requests to external endpoints.

Allowing GET unconditionally is a change in behavior, and it might make sense to have a second config setting that defines the regex match for GET requests.

The regex URLs themselves are a little confusing to me also, right now the path that needs to be matched in hardcoded on line 14 which limits the usefulness of this feature significantly (since only urls that match the hardcoded pattern can be allowed, which is particularly problematic if external urls are also in use because the URL regex applies to all POST and PUT requests and would block requests to external hosts that are currently allowed), is there any reason we need that vs having the user configure the paths that are allowed directly in the regex?

Reply:
resolving as we've removed this from the frontend